server/artifacts/artifact_server.go

```diff
@@ -186,41 +182,31 @@ func (a *ArtifactServer) returnArtifact(ctx context.Context, w http.ResponseWrit
 	if err != nil {
 		return err
 	}
-	tmp, err := ioutil.TempFile("/tmp", "artifact")
-	if err != nil {
-		return err
-	}
-	tmpPath := tmp.Name()
-	defer func() { _ = os.Remove(tmpPath) }()
-
-	err = driver.Load(art, tmpPath)
-	if err != nil {
-		return err
-	}
 
-	file, err := os.Open(filepath.Clean(tmpPath))
+	stream, err := driver.OpenStream(art)
 	if err != nil {
 		return err
 	}
 
 	defer func() {
-		if err := file.Close(); err != nil {
-			log.Fatalf("Error closing file[%s]: %v", tmpPath, err)
+		if err := stream.Close(); err != nil {
+			log.Fatalf("Error closing stream[%s]: %v", stream, err)
 		}
 	}()
 
-	stats, err := file.Stat()
-	if err != nil {
-		return err
-	}
-
-	contentLength := strconv.FormatInt(stats.Size(), 10)
-	log.WithFields(log.Fields{"size": contentLength}).Debug("Artifact file size")
-
 	key, _ := art.GetKey()
 	w.Header().Add("Content-Disposition", fmt.Sprintf(`filename="%s"`, path.Base(key)))
 
-	http.ServeContent(w, r, "", time.Time{}, file)
+	// Send the initial headers saying we're going to stream the response.
+	w.Header().Set("Transfer-Encoding", "chunked") // todo: do we need this?
```

Comment:
@alexec  were you saying you didn't think we need this?

Reply:
not needed